staging/src/k8s.io/component-base/metrics/prometheus/restclient/metrics.go

```diff
@@ -164,6 +182,9 @@ func init() {
 	legacyregistry.MustRegister(requestRetry)
 	legacyregistry.RawMustRegister(execPluginCertTTL)
 	legacyregistry.MustRegister(execPluginCertRotation)
+	legacyregistry.MustRegister(execPluginCalls)
```

Comment:
oh wow, nice catch. We might want to implement a static analysis tool to catch those

Reply:
Not sure how we would catch this via static analysis, but I think it would be easy enough to write a new e2e (in a separate PR) that checks the complete set of metrics registered against the API server (so that list would be updated every time someone added something new).

Reply:
> the complete set of metrics registered against the API server (so that list would be updated every time someone added something new)

this is a good issue for new contributors

Reply:
Yeah an e2e test will definitely be a good addition :+1:

I was thinking of having something static in order to catch metrics that are instantiated but not registered so that we don't have to rely on every component having up-to-date tests, although it would be great if they had.

Reply:
how do you do that? do you parse all files?

Reply:
yeah that's what I had in mind, but I don't know if it is feasible.

For instance that is already what we are doing to verify metrics stability: https://github.com/kubernetes/kubernetes/blob/master/test/instrumentation/stability-utils.sh